config/stack_mac.yaml

```diff
@@ -39,18 +39,18 @@ udunits:
 hdf5:
   build: YES
   version: 1.10.6
-  shared: NO
+  shared: YES
```

Comment:
Why shared? Static libraries are better in my opinion. I need to change this file every time I build hpc-stack on my mac.

Reply:
Do you have a really really strong argument against using shared libraries?

Reply:
`rpath` on macOS can mess up a lot, and there is always a danger that the wrong library gets picked if the runtime `LD_LIBRARY_PATH`/`DYLD_LIBRARY_PATH` doesn't match the one when the code is compiled. Or when homebrew decides that it wants to update a whole bunch of libraries accidentally (e.g. when the user wants to install something completely unrelated from homebrew). This doesn't happen with a static executable. But I won't insist on that, in the worst case I will change it every time or find another solution.

Reply:
Homebrew does not look at what we install via hpc-stack. So unless one "removes" a package from homebrew that was used for building a software in hpc-stack, the rpath issue should not arise.  
I have been able to use different gnu compiler versions (gcc-9.1, 9.2, 9.3) with the same hpc-stack built with gcc-9.1 from homebrew with no issues. 
I have had more issues with Xcode updates breaking my software than homebrew updates.